main.go

```diff
@@ -289,57 +315,66 @@ share the same seed as long as the indexes are different.
 
 		var seed string
 		var priv crypto.PrivKey
+		var peeringAddrs []peer.AddrInfo
+		var index int
 		var err error
 
-		credDir := os.Getenv("CREDENTIALS_DIRECTORY")
-		secretsDir := ddir
+		bitswap := cctx.Bool("bitswap")
+		dhtRouting := DHTRouting(cctx.String("dht-routing"))
+		seedPeering := cctx.Bool("seed-peering")
+		noLibp2p := !bitswap && dhtRouting == DHTOff && !seedPeering
```

Comment:
:thought_balloon:  `noLibp2p` setup feels overly complicated, and baloons the surface or permutations we need to test. 

In practice either user will run their own libp2p node for dht+bitswap, or they will run no libp2p at all, and delegate both retrieval and routing to remote HTTP endpoints.

@hacdias Could you tweak it to have an explicit, global `RAINBOW_LIBP2P=true|false` (with `true` being default) that controls when `SetupNoLibp2p` is used instead of `Setup[WithLibp2p]`? 

Then, inside `SetupNoLibp2p` just check if `bitswap`, `dht-routing`,  `seed-peering`, `libp2p` flags are set to `true`, and error otherwise.

We want end user's configuration to be simple:
`RAINBOW_LIBP2P=false RAINBOW_REMOTE_BACKENDS=http://trustless-gateway.link ./rainbow`

Reply:
> Then, inside `SetupNoLibp2p` just check if `bitswap`, `dht-routing`,  `seed-peering`, `libp2p` flags are set to `true`, and error otherwise.

Shouldn't they be `false` and `dht-routing=off`? If we want to check if they haven't been set, we need to do that before calling `SetupNoLibp2p`.

Reply:
I have pushed a commit with a global flag